src/components/Messages.tsx

```diff
@@ -0,0 +1,86 @@
+'use client'
+
+import {FC, useRef, useState} from "react";
+import {Message} from "@/lib/validations/messages"
+
+interface MessagesProps {
+    initialMessages: Message[],
+    sessionId: string
+}
+
+const Messages: FC<MessagesProps> = ({initialMessages, sessionId}) => {
+    const [messages, setMessages] = useState<Message[]>(initialMessages)
+    const scrollDownRef = useRef<HTMLDivElement | null>(null)
+
+    return <div aria-label='messages' className='message-scroll'>
+        <div ref={scrollDownRef}>
+        {messages.map((message, index) => {
+            const foo = messages[index - 1]?.senderId === messages[index].senderId
+            const classes = new ClassNames(sessionId === message.senderId, foo)
+
+            return (
+               <div key={listKey(message)} className={classes.div1}>
+                   <div className={classes.div2}>
+                       <span className={classes.span}>
+                           {message.text}{' '}
+                           <span className='message-date'>
+                               {message.timestamp}
+                           </span>
+                       </span>
+                   </div>
+               </div>
+           )
+        })}
+    </div>
+    </div>
+}
+
+const listKey = (message: Message) =>{
+    const {id, timestamp} = message
+    return `${id}-${timestamp}`
+}
+
+
+class ClassNames {
+    private readonly isCurrentUser: boolean
+    private readonly hasNextMessage: boolean
+
+    constructor(isCurrentUser: boolean, hasNextMessageFromSameUser: boolean) {
+        this.isCurrentUser = isCurrentUser
+        this.hasNextMessage = hasNextMessageFromSameUser
+    }
+
+    class(base: string, ifIsUserId: string, ifNotUserId: string | null = null){
+        const suffix = this.isCurrentUser ? ifIsUserId : ifNotUserId
+        if (!suffix){
+            return base
+        }
+        return `${base} ${suffix}`
+    }
+
+    get div1(): string{
```

Comment:
Move everything here into globals, get anal about the names so it clear to focus on the branching logis